core/templates/services/playthrough.service.ts

```diff
@@ -96,10 +98,16 @@ class CyclicStateTransitionsTracker {
     this.pathOfVisitedStates.push(destStateName);
   }
 
-  generateIssueCustomizationArgs(): CyclicStateTransitionsCustomizationArgs {
-    return {
-      state_names: {value: this.cycleOfVisitedStates}
-    };
+  // Return undefined if no cycle has been found.
+  generateIssueCustomizationArgs():
```

Comment:
Is it logically possible for this method to be called if no cycle has been found?

If it shouldn't be, then I would throw an exception in that case instead of returning undefined. Callers typically will not be expecting undefined as a return value.

Ditto elsewhere.

Reply:
Done, here `if` check is not needed, because we already check that condition before calling that function, hence removed and following by this we get rid of returing undefine.

Similar things happen with other, Ditto elsewhere.